stream-chat-android-ui-utils/src/main/kotlin/io/getstream/chat/android/uiutils/extension/Channel.kt

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.uiutils.extension
+
+import android.content.Context
+import androidx.annotation.PluralsRes
+import androidx.annotation.StringRes
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.extensions.getUsersExcludingCurrent
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.uiutils.constant.MessageType
+
+/**
+ * Returns channel's last regular or system message if exists.
+ * Deleted and silent messages, as well as messages from shadow-banned users, are not taken into account.
+ *
+ * @return Last message from the channel or null if it doesn't exist.
+ */
+public fun Channel.getPreviewMessage(currentUser: User?): Message? =
+    messages.asSequence()
+        .filter { it.createdAt != null || it.createdLocallyAt != null }
+        .filter { it.deletedAt == null }
+        .filter { !it.silent }
+        .filter { it.user.id == currentUser?.id || !it.shadowed }
+        .filter { it.type == MessageType.REGULAR || it.type == MessageType.SYSTEM }
+        .maxByOrNull { requireNotNull(it.createdAt ?: it.createdLocallyAt) }
+
+/**
+ * Returns the channel name if exists, or the list of member names if the channel is distinct.
+ *
+ * @param context The context to load string resources.
+ * @param currentUser The currently logged-in user.
+ * @param fallback The resource identifier of a fallback string if the [Channel] object lacks
+ * @param maxMembers The maximum number of members used to generate a name for a distinct channel.
+ * information to construct a valid display name string.
```

Comment:
Seems like we skipped a line here.
Related to the comment above: https://github.com/GetStream/stream-chat-android/pull/3941/files#r931005698